src/it/scala/com/databricks/spark/redshift/RedshiftIntegrationSuite.scala

```diff
@@ -386,6 +386,93 @@ class RedshiftIntegrationSuite extends IntegrationSuiteBase {
     }
   }
 
+  test("configuring compression on columns") {
+    val tableName = s"configuring_compression_on_columns_$randomSuffix"
+    try {
+      val metadata = new MetadataBuilder().putString("encoding", "LZO").build()
+      val schema = StructType(
+        StructField("x", StringType, metadata = metadata) :: Nil)
+      sqlContext.createDataFrame(sc.parallelize(Seq(Row("a" * 128))), schema).write
+        .format("com.databricks.spark.redshift")
+        .option("url", jdbcUrl)
+        .option("dbtable", tableName)
+        .option("tempdir", tempDir)
+        .mode(SaveMode.ErrorIfExists)
+        .save()
+      assert(DefaultJDBCWrapper.tableExists(conn, tableName))
+      val loadedDf = sqlContext.read
+        .format("com.databricks.spark.redshift")
+        .option("url", jdbcUrl)
+        .option("dbtable", tableName)
+        .option("tempdir", tempDir)
+        .load()
+      checkAnswer(loadedDf, Seq(Row("a" * 128)))
+      val encodingDF = sqlContext.read
+        .format("com.databricks.spark.redshift")
+        .option("url", jdbcUrl)
+        .option("dbtable",
+          s"""(SELECT "column", encoding FROM pg_table_def WHERE tablename='$tableName')""")
```

Comment:
This is a client-side error; I added `?loglevel=1` to the end of my JDBC URL in order to get the Redshift JDBC driver to perform more logging and used `DriverManager.setLogWriter(new PrintWriter(System.out))` to direct those logs to stdout.

This produced the following output:

```
SQLWarning: reason(Function "format_type(oid,integer)" not supported.) SQLState(01000) vendor code(0)
SQLWarning: reason(Function "pg_table_is_visible(oid)" not supported.) SQLState(01000) vendor code(0)
```

It looks like what's happening here is that we're trying to perform an `UNLOAD` on a leader-only table, which is unsupported: https://stackoverflow.com/questions/28719808/how-to-unload-pg-table-def-table-to-s3